pkgs/top-level/all-packages.nix

```
@@ -15553,6 +15553,12 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Security;
   };
 
+  buildTypstPackage = callPackage ../build-support/build-typst-package.nix {};
+
+  typstPackages = callPackage ./typst-packages.nix {
+    callPackage = lib.callPackageWith (pkgs // { inherit buildTypstPackage; });
+  };
```

```suggestion
  typstPackages = lib.packagesFromDirectoryRecursive {
    inherit callPackage;
    directory = ../development/typst-packages;
  }
```

Is this a good idea? It'd remove the need for `typst-packages.nix`, but I haven't seen `packagesFromDirectoryRecursive` in Nixpkgs before. (This is more of an open discussion and less of a review, though)

---

Have never used it before and happy to hear comments regarding it from other people

---

> but I haven't seen packagesFromDirectoryRecursive in Nixpkgs before.

This is the mechanism in place for automatically discovering packages in `pkgs/by-name`.

---

Then I suppose we don't need it? `typst-packages.nix` is an expression defined in the top-level and itself is not a discoverable "package" (as it does not follow the naming convention <package-name/package.nix>). Alternatively we could rename it to a different path but I leave it here because all other *-packages.nix are defined in the same directory. 

Also I was previously wondering how `pkgs/by-name` is discovered and joined to the `pkgs` set, and I guess this is it

---

> Then I suppose we don't need it?

It's up to you to decide how you want this to work. To be honest, I still don't understand everything in your PR so, I let you do some progress and then I'll test it out.

FYI: You can see an example of external use of `packagesFromDirectoryRecursive` in https://github.com/drupol/pkgs-by-name-for-flake-parts/

---

Ok, I think not as many other `*-packages.nix` are defined directly in the top-level directory and I decide to follow this pattern. The just pushed change removes `callPackage = lib.callPackageWith (pkgs // { inherit buildTypstPackage; });` as it is not needed.